src/App.tsx

```diff
@@ -1,61 +1,109 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types';
+import { getUserById, getNewTodoId } from './helpers/helpers';
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(todosWithUser);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!title.trim()) {
+      setTitleError(true);
+      setTitle('');
+
+      return;
+    }
```

Comment:
to pass the test, you need to remove this `return` and add `!title.trim()` on line 45

if i submit the form with empty field i will only see one error message
![image](https://github.com/mate-academy/react_add-todo-form/assets/96138503/6261db6d-38da-4bce-9a70-ac03bc5b2d87)